src/verifier/tdx/quote.rs

```diff
@@ -0,0 +1,255 @@
+            hex::encode(self.xfam),
+            hex::encode(self.mr_td),
+            hex::encode(self.mr_config_id),
+            hex::encode(self.mr_owner),
+            hex::encode(self.mr_owner_config),
+            hex::encode(self.rtmr_0),
+            hex::encode(self.rtmr_1),
+            hex::encode(self.rtmr_2),
+            hex::encode(self.rtmr_3),
+            hex::encode(self.report_data)
+        )
+    }
+}
+
+/// TD Quote Payload(Version 4)
+/// First 632 bytes of TD Quote
+/// Excluding the signature data attached at the end of the Quote.
+///
+/// Refer to: https://github.com/intel/SGXDataCenterAttestationPrimitives/blob/master/QuoteGeneration/quote_wrapper/common/inc/sgx_quote_4.h#L141
+#[repr(C)]
+#[derive(Debug, Pread)]
+pub struct Quote {
+    pub header: QuoteHeader,
+    pub report_body: ReportBody,
+}
+
+impl fmt::Display for Quote {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "TD Quote:\n{}\n{}\n", self.header, self.report_body)
+    }
+}
+
+pub fn parse_tdx_quote(quote_bin: &[u8]) -> Result<Quote> {
+    let quote_body = &quote_bin[..QUOTE_PAYLOAD_SIZE];
+    quote_body
+        .pread::<Quote>(0)
+        .map_err(|e| anyhow!("Parse TD quote failed: {:?}", e))
+}
+
+pub async fn ecdsa_quote_verification(quote: &[u8]) -> Result<()> {
+    let mut supplemental_data_size = 0u32; // mem::zeroed() is safe as long as the struct doesn't have zero-invalid types, like pointers
+    let mut supplemental_data: qvl::sgx_ql_qv_supplemental_t = unsafe { std::mem::zeroed() };
+    let mut quote_verification_result = qvl::sgx_ql_qv_result_t::SGX_QL_QV_RESULT_UNSPECIFIED;
+    let mut collateral_expiration_status = 1u32;
+
+    // call DCAP quote verify library to get supplemental data size
+    let dcap_ret = qvl::sgx_qv_get_quote_supplemental_data_size(&mut supplemental_data_size);
+    if qvl::quote3_error_t::SGX_QL_SUCCESS == dcap_ret
+        && std::mem::size_of::<qvl::sgx_ql_qv_supplemental_t>() as u32 == supplemental_data_size
+    {
+        debug!("\tInfo: TDX: sgx_qv_get_quote_supplemental_data_size successfully returned.");
+    } else {
+        if dcap_ret != qvl::quote3_error_t::SGX_QL_SUCCESS {
+            return Err(anyhow!(
+                "\tError: TDX: sgx_qv_get_quote_supplemental_data_size failed: {:#04x}",
+                dcap_ret as u32
+            ));
+        }
+
+        supplemental_data_size = 0u32;
+    }
+
+    // Set current time.
+    // This is the date that will be used to determine if any of the inputted collateral have expired.
+    let current_time: i64 = SystemTime::now()
+        .duration_since(SystemTime::UNIX_EPOCH)
+        .unwrap()
+        .as_secs()
+        .try_into()
+        .unwrap();
+
+    let p_supplemental_data = match supplemental_data_size {
+        0 => None,
+        _ => Some(&mut supplemental_data),
+    };
+
+    // Call DCAP quote verify library for quote verification
+    let dcap_ret = qvl::sgx_qv_verify_quote(
```

Comment:
I have checked the source code inside DCAP. SGX and TDX share a same entrance for quote verification. :D We can support SGX then with less effort